src/util/message.cpp

```diff
@@ -47,7 +47,7 @@ MessageVerificationResult MessageVerify(
         return MessageVerificationResult::ERR_PUBKEY_NOT_RECOVERED;
     }
 
-    if (!(CTxDestination(PKHash(pubkey)) == destination)) {
+    if (!(PKHash(pubkey) == *std::get_if<PKHash>(&destination))) {
```

Comment:
If IIUC this is a refactor.
So should be in it's own commit since the change does not concern 'Make WitnessUnknown members private' ?

Reply:
see https://github.com/bitcoin/bitcoin/pull/28246#discussion_r1318874324

It's not actually a refactor, but a fix for some very strange / edge case behavior with `std::variant` and `std::vector`